internal/service/streaming_processor.go

```diff
@@ -0,0 +1,571 @@
+package service
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/flexprice/flexprice/internal/domain/task"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/httpclient"
+	"github.com/flexprice/flexprice/internal/logger"
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// CircuitBreakerState represents the state of a circuit breaker
+type CircuitBreakerState int
+
+const (
+	CircuitBreakerClosed CircuitBreakerState = iota
+	CircuitBreakerOpen
+	CircuitBreakerHalfOpen
+)
+
+// CircuitBreaker implements the circuit breaker pattern for chunk processing
+type CircuitBreaker struct {
+	maxFailures  int
+	failureCount int
+	lastFailure  time.Time
+	resetTimeout time.Duration
+	state        CircuitBreakerState
+	mutex        sync.RWMutex
+}
+
+// NewCircuitBreaker creates a new circuit breaker
+func NewCircuitBreaker(maxFailures int, resetTimeout time.Duration) *CircuitBreaker {
```

Comment:
I recommend not to implement this custom written circuit breaker and rather in the first version -  just simplify without a circuit breaker or else use any existing library for circuit breaking.

We need to be sure of closing and opening logic of circuit breaking